v1/plugins/logs/plugin.go

```diff
@@ -372,13 +378,34 @@ func (c *Config) validateAndInjectDefaults(services []string, pluginsList []stri
 		return errors.New("invalid decision_log config, specify either 'buffer_size_limit_bytes' or 'max_decisions_per_second'")
 	}
 
+	if c.Reporting.BufferType == "" {
+		c.Reporting.BufferType = sizeBufferType
+	} else if c.Reporting.BufferType != eventBufferType && c.Reporting.BufferType != sizeBufferType {
+		return fmt.Errorf("invalid buffer type %q, expected %q or %q", c.Reporting.BufferType, eventBufferType, sizeBufferType)
+	}
+
+	if c.Reporting.BufferType == eventBufferType && c.Reporting.BufferSizeLimitBytes != nil {
+		return fmt.Errorf("invalid decision_log config, 'buffer_size_limit_bytes' isn't supported for the %v buffer type", eventBufferType)
+	}
+	if c.Reporting.BufferType == sizeBufferType && c.Reporting.BufferSizeLimitEvents != nil {
+		return fmt.Errorf("invalid decision_log config, 'buffer_size_limit_events' isn't supported for the %v buffer type", sizeBufferType)
+	}
+
 	// default the buffer size limit
-	bufferLimit := defaultBufferSizeLimitBytes
+	sizeBufferLimit := defaultBufferSizeLimitBytes
 	if c.Reporting.BufferSizeLimitBytes != nil {
-		bufferLimit = *c.Reporting.BufferSizeLimitBytes
+		sizeBufferLimit = *c.Reporting.BufferSizeLimitBytes
 	}
+	c.Reporting.BufferSizeLimitBytes = &sizeBufferLimit
 
-	c.Reporting.BufferSizeLimitBytes = &bufferLimit
+	eventBufferLimit := defaultBufferSizeLimitEvents
+	if c.Reporting.BufferSizeLimitEvents != nil {
+		if *c.Reporting.BufferSizeLimitEvents <= int64(0) {
+			return errors.New("invalid decision_log config, 'buffer_size_limit_entries' must be higher than 0")
```

Comment:
We could add a similar non-negative check for `c.Reporting.BufferSizeLimitBytes`?

Reply:
Good idea, added. Surprised it wasn't already there!